internal/evaluator/conftest_evaluator.go

```diff
@@ -445,11 +468,41 @@ func (c conftestEvaluator) Evaluate(ctx context.Context, target EvaluationTarget
 						}
 					}
 
-					// Extract short name from the rule head
-					shortName := ruleRef.Head.Name.String()
+					// Try to extract code from rule body first, fallback to rule name
+					code := ""
+
+					// Look for result assignment with code in the rule body
+					if ruleRef.Body != nil {
+						for _, expr := range ruleRef.Body {
+							if expr.IsAssignment() {
+								// Check if this is a result assignment
+								if len(expr.Operands()) >= 2 {
+									// Look for result := { "code": "...", ... }
+									if term, ok := expr.Operands()[1].Value.(ast.Object); ok {
```

Comment:
You could use early returns here to flatten the syntax. Something like:
```suggestion
							if !expr.IsAssignment() {
								// Check if this is a result assignment
								continue
							}
							if len(expr.Operands()) < 2 {
								continue
							}
							
							// Look for result := { "code": "...", ... }
							if term, ok := expr.Operands()[1].Value.(ast.Object); !ok {
								continue
							}
```
and so on